crates/wdl-analysis/src/handlers/hover.rs

```diff
@@ -248,14 +263,30 @@ fn resolve_hover_by_context(
             let Some(scope) = document.find_scope_by_position(parent_node.span().start()) else {
                 return Ok(None);
             };
+
             let mut ctx = TypeEvalContext { scope, document };
             let mut evaluator = ExprTypeEvaluator::new(&mut ctx);
             let target_type = evaluator
                 .evaluate_expr(&expr)
                 .unwrap_or(crate::types::Type::Union);
 
             let (member_ty, documentation) = match target_type {
-                Type::Compound(CompoundType::Struct(s), _) => {
+                Type::TypeNameRef(CustomType::Enum(e)) => {
+                    if e.variants().iter().any(|text| text == member.text()) {
+                        let content = format!(
+                            "```wdl\n{}.{}[{}]\n```",
+                            e.name(),
+                            member.text(),
+                            e.inner_value_type()
+                        );
+                        return Ok(Some(content));
+                    }
+                    (None, None)
+                }
+                Type::TypeNameRef(CustomType::Struct(_)) => {
+                    todo!("handle struct member access via `TypeNameRef`")
```

Comment:
Same comment as above: I don't think the LSP should handle these as they aren't valid expressions anywhere in WDL.